### DISCUSSION TITLE
Downloading all the files per recording.

### DISCUSSION THREAD

**Discussion by AndreaMele** [hyunbinseo/zoom-rec-dl#27] [General]
Hello, I hope I didn't miss this somewhere.

I'm able to download the videos for each of my recordings, but I would love to be able to grab the audio only file, and the transcript.

![image](https://github.com/user-attachments/assets/0b823aee-3616-4f90-8d3a-05526254508c)

**Comment by hyunbinseo** [comment]
There is no feature to select specific file types (or extension).

If you are using npm (and not pnpm), you can modify this part of the code:

https://github.com/hyunbinseo/zoom-rec-dl/blob/v2.1.10/src/index.ts#L173-L178

Are you interested in this workaround?

**Comment by thomasexner** [comment]
Hallo:
I would be very interested in this. However, I tried to change the lines but I did not find a way to get the .vtt files. And even if the .m4a files should be downloaded, it looks like that the script is currently downloading the .mp4 files twice since it is the only file in the response from querying the URL. This might be something caused by a recent change in the Zoom interface.
Thanks a lot.

**Reply by hyunbinseo** [reply]
Multiple mp4 files can be downloaded at once. For example:

```
GMT20250401-053208_Recording_gvo_1280x720.mp4 
GMT20250401-053208_Recording_1920x1020.mp4 
GMT20250401-053208_Recording_as_1920x1020.mp4 
GMT20250401-053208_Recording_avo_1280x720.mp4 
GMT20250401-053208_Recording_gallery_1920x1020.mp4
```

Is the vtt file downloaded from the Zoom web interface accessed from the browser?

**Reply by thomasexner** [reply]
![Screenshot 2025-04-02 081641](https://github.com/user-attachments/assets/ba4da412-248c-4ff1-b95e-8126c786d4c7)
Thanks for the quick reply. When I use the URL, on which I can access the shared Zoom video (as included in the urls.txt), I can download three files by clicking on the button on the top right (see attached image - "Herunterladen"): video (.mp4), audio file (.m4a) and transcript (.vtt). Thus, the .vtt is not automatically downloaded if this is what you asked about, but neither is the .mp4 (It can be watched embedded in the browser window).

**Reply by hyunbinseo** [reply]
Could you share a reproduction that I can check?

- all `.mp4` and `.m4a` files should be downloaded as of `2.1.10`
- `.vtt` is not downloaded by default and it should be added to the regex as [commented above](https://github.com/hyunbinseo/zoom-rec-dl/discussions/27#discussioncomment-10699010)

**Reply by thomasexner** [reply]
Sorry, but this is exactly what I tried. The .m4a file is not downloaded using the latest version and adding .vtt in the regex doesn't change anything since there is no .vtt in the playInfo object (and neither is the .m4a file). I assume that the fetch call would need to be adapted but I do not know enough about zoom and your code to do this.

**Reply by hyunbinseo** [reply]
By reproduction, I meant a public Zoom link that I can check on.

If the following is the case, it might take longer to implement it:

> since there is no .vtt in the playInfo object (and neither is the .m4a file)

**Reply by thomasexner** [reply]
Now I understand. Here is the link to a short test video:  https://us06web.zoom.us/rec/share/IYdPZxZOEd6Vb3J4m1k6WFWcMLIq_c3b_pnJyEBmYQ2J6KqYPl0jlkKedlddahLv.HH4r0PLUg_LgsyPN?pwd=1TDaa4lqrXGcvchkCUrkXcV_7iy09juM

And I also attached a screenshot of the output of zoom-rec-dl@latest showing that the .mp4 file was downloaded twice but not the also available .m4a file.
![Screenshot 2025-04-02 105541](https://github.com/user-attachments/assets/d7c89136-6743-4447-8f90-07b8e43e7b36)

**Reply by hyunbinseo** [reply]
Turns out it is midterm season in university. Could this wait for 1-2 weeks?

**Reply by thomasexner** [reply]
Absolutely no problem. Thanks for looking into this.

But I also have another related question, perhaps somebody here knows an answer for: Is there a way to get all the links of shared meetings? At the moment, I have to check them all individually in a browser and copy the links into the urls.txt file. This almost takes as much time as downloading it directly from the web interface. Would be much nicer if we could get this in a summary file but the export doesn't provide this information.

**Reply by hyunbinseo** [reply]
Our Zoom provider is actually providing us with the shared meetings URL, which is why it is not implemented.

Is it similar to the following comment?

- https://github.com/hyunbinseo/zoom-rec-dl/discussions/17#discussioncomment-6336748
